File: midterm-Lab3/MP17/MP17_LongestText.java
import java.io.*;
import java.util.*;

/*
MP17 – Find the longest text entry in dataset
This program reads a CSV file and finds the row that contains
the longest text entry based on character length.
*/

public class MP17_LongestText {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        // Ask user for dataset path
        System.out.print("Enter dataset file path: ");
        String filePath = input.nextLine();

        String longestText = "";   // Stores the longest row

        try {

            // Read the CSV file
            BufferedReader reader = new BufferedReader(new FileReader(filePath));
            String line;

            // Loop through every row
            while ((line = reader.readLine()) != null) {

                // Compare length of current row with longest stored row
                if (line.length() > longestText.length()) {
                    longestText = line;
                }

            }

            reader.close();

            // Display result
            System.out.println("\nLongest Text Entry:");
            System.out.println(longestText);
            System.out.println("\nLength: " + longestText.length() + " characters");

        } catch (Exception e) {

            System.out.println("Error reading file.");

        }

    }
}

File: midterm-Lab3/MP17/MP17_LongestText.js
const fs = require("fs");
const readline = require("readline");

/*
MP17 – Find the longest text entry in dataset
This program reads a CSV file and finds the row
with the longest text entry.
*/

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter dataset file path: ", function(path) {
    try {
        const data = fs.readFileSync(path, "utf8");

        // Remove extra line ending characters
        const rows = data.split("\n").map(row => row.trimEnd());

        let longestText = "";

        rows.forEach(row => {
            if (row.length > longestText.length) {
                longestText = row;
            }
        });

        console.log("\nLongest Text Entry:");
        console.log(longestText);
        console.log("\nLength:", longestText.length, "characters");

    } catch (err) {
        console.log("Error reading file.");
    }

    rl.close();
});